<div class="reveal">
    <div class="slides">
        <section data-background-color="var(--pb-orange-color)">
            <h2>Promise.resolve() and Promise.reject()</h2>
            <p class="fst-italic">The <b>Promise.resolve()</b> method "resolves" a given value to a Promise. If the value is a promise, that promise is returned; if the value is a thenable, <b>Promise.resolve()</b> will call the then() method with two callbacks it prepared; otherwise the returned promise will be fulfilled with the value.</p>
            <p class="text-end">
                <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve" target="blank">mdn web docs</a>
            </p>
            <p class="fst-italic">The Promise.reject() method returns a Promise object that is rejected with a given reason.</p>
            <p class="text-end">
                <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject" target="blank">mdn web docs</a>
            </p>
        </section>
        <section>
            <h3 class="pb-title">Promise.resolve()</h3>
            <section data-auto-animate>
                <pre data-id="code-animation">
                    <code class="language-js" data-trim data-line-numbers>
                        Promise.resolve('resolve value');
                    </code>
                </pre>
            </section>
            <section>
                <p>
                    Sometimes in tests we have to immediately resolve Promise with a given data.
                </p>
            </section>
        </section>
        <section>
            <h3 class="pb-title">Promise.resolve()</h3>
            <section data-auto-animate>
                <pre data-id="code-animation">
                    <code class="language-js" data-trim data-line-numbers>
                        Promise.resolve('resolve value').then(data => {
                            console.log(data); // logs 'resolve value'
                        });
                    </code>
                </pre>
            </section>
            <section>
                <p>
                    Promise.resolve() returns Promise and we can intercept data using then().
                </p>
            </section>
        </section>
        <section>
            <h3 class="pb-title">Promise.reject()</h3>
            <section data-auto-animate>
                <pre data-id="code-animation">
                    <code class="language-js" data-trim data-line-numbers>
                        Promise.reject(new Error('Error message'));
                    </code>
                </pre>
            </section>
            <section>
                <p>
                    Sometimes in tests we have to immediately reject Promise with a given Error message.
                </p>
            </section>
        </section>
        <section>
            <h3 class="pb-title">Promise.reject()</h3>
            <section data-auto-animate>
                <pre data-id="code-animation">
                    <code class="language-js" data-trim data-line-numbers>
                        Promise.reject(new Error('Error message')).catch(error => {
                            console.error(error); // logs error
                        });
                    </code>
                </pre>
                <button class="fragment below pb-next" type="button" onclick="nextBranch('async-5-1-1')">2 go</button>
            </section>
            <section>
                <p>
                    Promise.reject() returns Promise and we can intercept error using catch().
                </p>
            </section>
        </section>
    </div>
</div>